src/lib/helpers/componentCollectionContext.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Context for registering child components
+ */
+import { get, writable, type Writable } from 'svelte/store';
+import { uniqueContext } from './uniqueContext';
+import type { SvelteHTMLActionType } from './useActions';
+
+export function componentCollectionContext() {
+	const initialContext = uniqueContext<Writable<Array<HTMLElement>>>();
+
+	const createContext = () => {
+		const store = writable<HTMLElement[]>([]);
+		initialContext.setContext(store);
+		return store;
+	};
+
+	return { ...initialContext, createContext };
```

Comment:
Should we return setContext alongside createContext? That could cause confusion IMO